File: src/main/java/com/itsci/mju/maebanjumpen/skilltype/service/SkillLevelTierService.java
package com.itsci.mju.maebanjumpen.skilltype.service;

import com.itsci.mju.maebanjumpen.skilltype.dto.SkillLevelTierDTO;
import java.util.List;

public interface SkillLevelTierService {
    List<SkillLevelTierDTO> getAllSkillLevelTiers();
    SkillLevelTierDTO getSkillLevelTierById(Integer id);
    SkillLevelTierDTO createSkillLevelTier(SkillLevelTierDTO skillLevelTierDTO);
    SkillLevelTierDTO updateSkillLevelTier(Integer id, SkillLevelTierDTO skillLevelTierDTO);
    void deleteSkillLevelTier(Integer id);

}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/skilltype/service/SkillLevelTierService.kt
package com.itsci.mju.maebanjumpen.skilltype.service

import com.itsci.mju.maebanjumpen.skilltype.dto.SkillLevelTierDTO
import com.itsci.mju.maebanjumpen.skilltype.request.CreateSkillTierRequest
import com.itsci.mju.maebanjumpen.skilltype.request.UpdateSkillTierRequest

interface SkillLevelTierService {
    fun listAllSkillLevelTiers(): List<SkillLevelTierDTO>
    fun getSkillLevelTierById(id: Long): SkillLevelTierDTO?
    fun createSkillLevelTier(request: CreateSkillTierRequest): SkillLevelTierDTO
    fun updateSkillLevelTier(request: UpdateSkillTierRequest): SkillLevelTierDTO
    fun deleteSkillLevelTier(id: Long): Boolean
}

